### PR TITLE
Improves contribution guideline

### PR DESCRIPTION
**Pull request by vivonk**
preview of changes can be seen [here](https://github.com/vivonk/scancode-toolkit/blob/improve-contribution-guideline/CONTRIBUTING.rst)
Things that are done

- Refactored links to clickable texts which is something [this](https://github.com/nexB/scancode-toolkit/pulls)
- Added extra git commands for general help
- Explained some sentences in bit deep
- Added external links of useful resources to give a clear context

Things that are left to do

1.  Script/document to support conda environment
2.  Add some lines about how to merge with [master branch](https://github.com/nexB/scancode-toolkit) after solving a issue or keep updated to the master branch of forked repository from the Git Hub web version

Signed-off-by: vivonk <nirmalsarswat400@gmail.com>

### CODE REVIEW

**Comment by vivonk** [comment]
@pombredanne can you review please

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -5,28 +5,27 @@ Contributing
...
+For writing good commit messages and comments with markdown, read `this guide <https://guides.github.com/features/mastering-markdown/>`_ for once
 
 
 
 Feature requests and feedback
 =============================
 
-To send feedback or ask a question file an issue at
-https://github.com/scancode/scancode-toolkit/issues
+To send feedback or ask a question file an issue at `issues <https://github.com/scancode/scancode-toolkit/issues>`_
```
Comment (haikoschol) [inline CONTRIBUTING.rst]: Adding "at issues" to the end seems a bit redundant. How about making "file an issue" the link?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -47,13 +46,13 @@ For other questions, discussions, and chats, we have:
...
 
-- a mailing list at https://lists.sourceforge.net/lists/listinfo/aboutcode-discuss
+- a mailing list at `sourceforge <https://lists.sourceforge.net/lists/listinfo/aboutcode-discuss>`_
 
 
 Bug reports
 ===========
 
-When reporting a bug at https://github.com/nexb/scancode-toolkit/issues please
+When reporting a bug at `issues <https://github.com/nexb/scancode-toolkit/issues>`_ please
```
Comment (haikoschol) [inline CONTRIBUTING.rst]: Same as above: "reporting a bug" could be the link text.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (vivonk) [reply CONTRIBUTING.rst]: sure

File: CONTRIBUTING.rst (hunk 2 above)
Comment (vivonk) [reply CONTRIBUTING.rst]: sure

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -137,15 +149,20 @@ To set up ScanCode for local development:
...
-7. Commit your changes and push your branch to your GitHub fork::
+7. Check the status of your local repository before commit, regarding files changed::
+    
+    git status
 
-    git add .
+
+8. Commit your changes and push your branch to your GitHub fork::
+
+    git add <file-changed-1> <file-changed-2> <file-changed-3> / git add .
```
Comment (haikoschol) [inline CONTRIBUTING.rst]: I like how this discourages `git add .`. But the way it is written now can be misunderstood as having to type the whole line, including ` / git add .`. Should we remove that completely?

Comment (vivonk) [reply CONTRIBUTING.rst]: Consider you are working on an issue which is likely to get solved by 3-4 commits , in this case git add should consider only those file which are about to commit right now.

Comment (haikoschol) [reply CONTRIBUTING.rst]: I think we are in agreement here that `git add .` should be avoided. I was just wondering whether we should remove it completely from this line.

Comment (vivonk) [reply CONTRIBUTING.rst]: correct

**Pull request closed without merging**